can't add new files (created on chain-tip) to downloader

```
[INFO] [08-05|13:20:26.336] Timings: Post-Forkchoice                 Prune TxLookup=70ms Prune Execution=185ms
[INFO] [08-05|13:20:38.328] head updated                             hash=0x68a0d60a174786a78d3eca35bac49e569bf9f176c023c17b0dc14a0d467b3f2b number=23075125 txnum=2971505847 age=3s execution=592.306859ms mgas/s=42.72 average mgas/s=40.69 commit=160.213269ms alloc=17.4GB sys=30.2GB
[INFO] [08-05|13:20:38.514] Timings: Post-Forkchoice                 Prune TxLookup=79ms Prune Execution=51ms
[INFO] [08-05|13:20:49.915] head updated                             hash=0xba63fb4cbadc8925dfb9e733876fad756dbb354c351aca77c0a8f0569f825a38 number=23075126 txnum=2971506087 age=2s execution=580.780842ms mgas/s=37.61 average mgas/s=40.67 commit=135.727133ms alloc=18.2GB sys=30.2GB
[INFO] [08-05|13:20:50.071] Timings: Post-Forkchoice                 Prune TxLookup=76ms
[INFO] [08-05|13:21:01.405] head updated                             hash=0x39f884658fb4c9fb63332cd7b4d8c98fb0ce4661ff35475e38ed72ad1d31815f number=23075127 txnum=2971506199 age=2s execution=157.852746ms mgas/s=39.78 average mgas/s=40.66 commit=55.477182ms alloc=18.8GB sys=30.2GB
[INFO] [08-05|13:21:01.857] Timings: Post-Forkchoice                 Prune TxLookup=68ms Prune Execution=282ms
[INFO] [08-05|13:21:14.851] head updated                             hash=0xd58fff765931d3ac1dc66f5bd79ba37a67959e142fdaaeb826422a599d74e1b2 number=23075128 txnum=2971506681 age=3s execution=1.131633528s mgas/s=39.26 average mgas/s=40.65 commit=230.970364ms alloc=19.9GB sys=30.2GB
[INFO] [08-05|13:21:15.241] Timings: Post-Forkchoice                 Prune TxLookup=67ms Prune Execution=231ms
[INFO] [08-05|13:21:18.946] P2P                                      app=caplin peers=126
[INFO] [08-05|13:21:25.804] head updated                             hash=0x15f7800f402fd4bb1fd0d9a99916a9c9cfb2d12e9c571555547deae12930d177 number=23075129 txnum=2971506891 age=2s execution=480.602336ms mgas/s=40.32 average mgas/s=40.65 commit=130.104512ms alloc=20.4GB sys=30.2GB
[INFO] [08-05|13:21:26.185] Timings: Post-Forkchoice                 Prune TxLookup=80ms Prune Execution=208ms
[INFO] [08-05|13:21:36.030] [Antiquary] Finished Antiquating blobs   app=caplin from=12230000 to=12260000
panic: chunk size cannot be changed for existing Torrent

goroutine 40609 [running]:
github.com/anacrolix/torrent.(*Torrent).MergeSpec(0xc110076808, 0xc191d0d138)
	github.com/anacrolix/torrent@v1.58.2-0.20250720014114-dda1d97c6a22/client.go:1565 +0x506
github.com/anacrolix/torrent.(*Client).AddTorrentSpec(0xc191d0d308?, 0xc10f04d8c0)
	github.com/anacrolix/torrent@v1.58.2-0.20250720014114-dda1d97c6a22/client.go:1528 +0xd7
github.com/erigontech/erigon-db/downloader.(*Downloader).addTorrentSpec(0xc04fdc4e00, 0xc3a2fde8d0?, {0xc3a2fde8d0, 0x23})
	github.com/erigontech/erigon-db@v0.0.0-00010101000000-000000000000/downloader/util.go:340 +0xa7
github.com/erigontech/erigon-db/downloader.(*Downloader).AddNewSeedableFile(0xc04fdc4e00, {0x3b401f0, 0xc3c1fc9040}, {0xc3a2fde8d0, 0x23})
	github.com/erigontech/erigon-db@v0.0.0-00010101000000-000000000000/downloader/downloader.go:861 +0x35a
github.com/erigontech/erigon-db/downloader.(*GrpcServer).Add(0xc000092a20, {0x3b401f0?, 0xc110604320?}, 0xc69292c300)
	github.com/erigontech/erigon-db@v0.0.0-00010101000000-000000000000/downloader/downloader_grpc_server.go:95 +0x2e5
github.com/erigontech/erigon-lib/direct.(*DownloaderClient).Add(0xc10f039b20?, {0x3b401f0?, 0xc110604320?}, 0xc4e309e680?, {0xc0011a26f0?, 0x16?, 0xc0011a2708?})
	github.com/erigontech/erigon-lib@v0.0.0-00010101000000-000000000000/direct/downloader_client.go:38 +0x25
github.com/erigontech/erigon/cl/antiquary.(*Antiquary).antiquateBlobs(0xc24326ca00)
	github.com/erigontech/erigon/cl/antiquary/antiquary.go:429 +0x819
github.com/erigontech/erigon/cl/antiquary.(*Antiquary).Loop(0xc24326ca00)
	github.com/erigontech/erigon/cl/antiquary/antiquary.go:265 +0x173c
github.com/erigontech/erigon/cmd/caplin/caplin1.RunCaplinService.func3()
	github.com/erigontech/erigon/cmd/caplin/caplin1/run.go:412 +0xe7
created by github.com/erigontech/erigon/cmd/caplin/caplin1.RunCaplinService in goroutine 350
	github.com/erigontech/erigon/cmd/caplin/caplin1/run.go:405 +0x334e
```

--- a/db/downloader/downloader.go
+++ b/db/downloader/downloader.go
@@ -882,15 +882,15 @@ func (d *Downloader) AddNewSeedableFile(ctx context.Context, name string) error 
 	// if we don't have the torrent file we build it if we have the .seg file
 	_, err := BuildTorrentIfNeed(ctx, name, d.SnapDir(), d.torrentFS)
 	if err != nil {
-		return fmt.Errorf("AddNewSeedableFile: %w", err)
+		return fmt.Errorf("building metainfo for new seedable file: %w", err)
 	}
-	ts, err := d.torrentFS.LoadByName(name)
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	// The above BuildTorrentIfNeed should put the metainfo in the right place for name.
+	// addPreverifiedTorrent is the correct wrapper to check for existing torrents in the client.
+	_, err = d.addPreverifiedTorrent(g.None[metainfo.Hash](), name)
 	if err != nil {
-		return fmt.Errorf("AddNewSeedableFile: %w", err)
-	}
-	_, _, err = d.addTorrentSpec(ts, name)
-	if err != nil {
-		return fmt.Errorf("addTorrentSpec: %w", err)
+		return fmt.Errorf("adding torrent: %w", err)
 	}
 	return nil
 }

--- a/go.mod
+++ b/go.mod
@@ -31,7 +31,7 @@ require (
 	github.com/anacrolix/go-libutp v1.3.2
 	github.com/anacrolix/log v0.16.1-0.20250526073428-5cb74e15092b
 	github.com/anacrolix/missinggo/v2 v2.10.0
-	github.com/anacrolix/torrent v1.58.2-0.20250808032922-b6e9e69c96b4
+	github.com/anacrolix/torrent v1.58.2-0.20250811011913-5c778813ff6d
 	github.com/c2h5oh/datasize v0.0.0-20231215233829-aa82cc1e6500
 	github.com/cenkalti/backoff/v4 v4.2.1
 	github.com/consensys/gnark-crypto v0.18.0

